src/SriServiceProvider.php

```diff
@@ -29,39 +30,89 @@ public function boot()
         ]);
 
         Blade::directive('mixSri', function (string $path, bool $crossOrigin = false) {
+            $path = $this->clean_path($path);
+
             if (Str::startsWith($path, ['http', 'https', '//'])) {
                 $href = $path;
             } else {
                 $href = mix($path);
             }
 
-            $integrity = SriFacade::html($path, $crossOrigin);
-
-            if (Str::endsWith($path, 'css')) {
-                return "<link href='{$href}' rel='stylesheet' {$integrity}>";
-            } elseif (Str::endsWith($path, 'js')) {
-                return "<script src='{$href}' {$integrity}></script>";
-            } else {
-                throw new \Exception('Invalid file');
-            }
+            return $this->generate($path, $href, $crossOrigin);
         });
 
         Blade::directive('assetSri', function (string $path, bool $crossOrigin = false) {
+            $path = $this->clean_path($path);
+
             if (Str::startsWith($path, ['http', 'https', '//'])) {
                 $href = $path;
             } else {
                 $href = asset($path);
             }
 
-            $integrity = SriFacade::html($path, $crossOrigin);
-
-            if (Str::endsWith($path, 'css')) {
-                return "<link href='{$href}' rel='stylesheet' {$integrity}>";
-            } elseif (Str::endsWith($path, 'js')) {
-                return "<script src='{$href}' {$integrity}></script>";
-            } else {
-                throw new \Exception('Invalid file');
-            }
+            return $this->generate($path, $href, $crossOrigin);
         });
     }
+
+    /**
+     * Remove apostrophe and quotation mark.
```

Comment:
I would rename the function to something more expressive. That way we can remove the whole docblock as it won't be needed anymore.

A name that you could use is: `removeQuotes` (as it encapsulates both single and double quotes).